server/cli/launch.py

```diff
@@ -129,103 +200,61 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def parse_engine_args(
-    embedding,
-    obs_names,
-    var_names,
-    max_category_items,
-    diffexp_lfc_cutoff,
-    experimental_annotations,
-    experimental_annotations_file,
-    experimental_annotations_output_dir,
-    backed,
-    disable_diffexp,
-    experimental_annotations_ontology,
-    experimental_annotations_ontology_obo
-):
-    annotations_file = experimental_annotations_file if experimental_annotations else None
-    annotations_output_dir = experimental_annotations_output_dir if experimental_annotations else None
-    annotations_cell_ontology_enabled = experimental_annotations and (
-        experimental_annotations_ontology or bool(experimental_annotations_ontology_obo)
+def launch_args(func):
+    @annotation_args
+    @config_args
+    @dataset_args
+    @server_args
+    @click.option("--dataroot", default=None, metavar="<data files root>")
```

Comment:
the help needs to be better.   Try running `cellxgene launch --help` -- there is insufficient explanation for the new dataroot option.

Also, I'm not sure it makes sense to specify both a data file and a dataroot.  Shouldn't they be exclusive?   The CLI options allows both.

Recommendation:
* make them exclusive modes
* accept only a single path (currently known as `data` on the command line)
* `--datamode` means treat the path like a directory, and defaults OFF

Also, for this PR, if we are not yet ready to support dataroot (and I think we are not quite there), I would tag the `--dataroot` option as `hidden=True`

Reply:
And on the help text, if you need suggestions, Lia is fantastic at clear wording.

Reply:
Fixed.  They are now exclusive, and the dataroot option is hidden.  I've renamed the "data" option to "datapth" since that makes a bit more sense.   So you either specify a dataroot or a datapath, but never both.

Reply:
Lia helped provide text for the help message.